packages/react-hooks/src/useSetInterval.ts

```diff
@@ -0,0 +1,30 @@
+import { useConst } from './useConst';
+import { useEffect } from 'react';
+
+/**
+ * Returns a self disposing setTimeout function.
+ */
+export const useSetInterval = () => {
+  const intervalIds = useConst<number[]>([]);
+
+  useEffect(
+    () => () => {
+      for (const id of intervalIds) {
+        clearInterval(id);
+      }
+      intervalIds.splice(0);
+    },
+    [],
+  );
+
+  return (func: () => void, duration: number) => {
+    const id = (setInterval(() => {
+      intervalIds.splice(intervalIds.indexOf(id), 1);
+      func();
+    }, duration) as unknown) as number;
```

Comment:
not sure why you need to cast `as unknown` here?

Reply:
The types for setInterval get messed up due to presence of `@types/node` in the package. In Node it returns some other type, not a number like in the browser.